bigtable/admin.go

```diff
@@ -547,23 +547,60 @@ func (ac *AdminClient) TableInfo(ctx context.Context, table string) (*TableInfo,
 	return ti, nil
 }
 
-// SetGCPolicy specifies which cells in a column family should be garbage collected.
-// GC executes opportunistically in the background; table reads may return data
-// matching the GC policy.
-func (ac *AdminClient) SetGCPolicy(ctx context.Context, table, family string, policy GCPolicy) error {
+type gcPolicySettings struct {
+	ignoreWarnings bool
+}
+
+// GCPolicyOption is the interface to change GC policy settings
+type GCPolicyOption interface {
+	apply(s *gcPolicySettings)
+}
+
+type warnings struct{ ignore bool }
+
+func (w warnings) apply(s *gcPolicySettings) {
+	s.ignoreWarnings = w.ignore
+}
+
+// IgnoreWarnings returns a gcPolicyOption that ignores safety checks when modifying the column families
+func IgnoreWarnings() GCPolicyOption {
+	return warnings{ignore: true}
+}
+
+func (ac *AdminClient) setGCPolicy(ctx context.Context, table, family string, policy GCPolicy, opts ...GCPolicyOption) error {
 	ctx = mergeOutgoingMetadata(ctx, ac.md)
 	prefix := ac.instancePrefix()
+
+	s := gcPolicySettings{}
+	for _, opt := range opts {
+		if opt != nil {
+			opt.apply(&s)
+		}
+	}
 	req := &btapb.ModifyColumnFamiliesRequest{
 		Name: prefix + "/tables/" + table,
 		Modifications: []*btapb.ModifyColumnFamiliesRequest_Modification{{
 			Id:  family,
 			Mod: &btapb.ModifyColumnFamiliesRequest_Modification_Update{Update: &btapb.ColumnFamily{GcRule: policy.proto()}},
 		}},
+		IgnoreWarnings: s.ignoreWarnings,
 	}
 	_, err := ac.tClient.ModifyColumnFamilies(ctx, req)
 	return err
 }
 
+// SetGCPolicy specifies which cells in a column family should be garbage collected.
+// GC executes opportunistically in the background; table reads may return data
+// matching the GC policy.
+func (ac *AdminClient) SetGCPolicy(ctx context.Context, table, family string, policy GCPolicy) error {
+	return ac.SetGCPolicyWithOptions(ctx, table, family, policy)
+}
+
+// SetGCPolicyWithOptions is similar to SetGCPolicy but allows passing options
+func (ac *AdminClient) SetGCPolicyWithOptions(ctx context.Context, table, family string, policy GCPolicy, opts ...GCPolicyOption) error {
+	return ac.setGCPolicy(ctx, table, family, policy, opts...)
+}
+
```

Comment:
I don't think you need another function for this. You can add `opts ...GcPolicyOption` to the previous function?

Reply:
This would lead to apidiff workflow failure
I had discussed similar scenario with Cody. There is a decent chance a change like this breaks someone. In Go, and just best practices, users should define an interface for dependencies they take on. This allows them to more easily test/mock interactions of the dependency. A change like this would break their interface if they had one that included the SetGCPolicy method as our client would no longer fulfil it.

Reply:
Ahh, I see. Makes sense then.